model/task/db.go

```diff
@@ -2839,6 +2840,16 @@ func SetHasAnnotations(taskId string, execution int) error {
 	return errors.Wrapf(err, "marking task '%s' as having annotations", taskId)
 }
 
+// SetNumNextTaskDispatches sets the next task count for a task.
+func SetNumNextTaskDispatches(taskId string, execution int, count int) error {
+	err := UpdateOne(
+		ByIdAndExecution(taskId, execution),
+		bson.M{"$set": bson.M{
+			NumNextTaskDispatchesKey: count,
+		}})
```

Comment:
nit: don't forget to update the in-memory task itself. I don't think it's that important though if you want to ignore this.